### PR TITLE
[Go] Client Models using AllOf, OneOf, or AnyOf Correctly Include time and os imports (Attempt 2)

### PR DESCRIPTION
**Pull request by icubbon**
Addresses https://github.com/OpenAPITools/openapi-generator/issues/13959

Currently, if a Model is an allOf the time and os imports are not correctly added to the generated file. This was introduced recently with a fix to not include those imports when the model is a composedSchema #13833. The logic in that fix was just slightly off as an allOf should be treated the same as a standard model.

If a model is an AllOf or does not have any composed schemas at all, the sub-models are in-lined defined in the struct. In this case, the standard logic of including the time and os imports apply.

If a model is a OneOf or AnyOf, the sub-models are included as pointers to the defined model. In this case, do not include those items in the logic of including time and os imports.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@antihax @grokify @kemokemo @jirikuncar @ph4r5h4d @wing328

### CODE REVIEW

**Comment by wtrocki** [comment]
Can you add unit test from my PR.
There are couple duplicate issues that I have closed.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractGoCodegen.java [hunk 1]
```diff
@@ -671,23 +664,11 @@ public ModelsMap postProcessModels(ModelsMap objs) {
...
                 imports.add(createMapping("import", "fmt"));
             }
 
-            // if oneOf contains "time.Time" type
-            if (!addedTimeImport && model.oneOf != null && model.oneOf.contains("time.Time")) {
-                imports.add(createMapping("import", "time"));
-            }
-
             // if oneOf contains "null" type
-            if (model.oneOf != null && !model.oneOf.isEmpty() && model.oneOf.contains("nil")) {
```
Comment (wtrocki) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractGoCodegen.java:685]: Why this was removed?

**Comment by icubbon** [comment]
> 

Do you mean the example added to `modules/openapi-generator/src/test/resources/3_0/go/petstore-with-fake-endpoints-models-for-testing-with-http-signature.yaml`? I don't see a unit test in https://github.com/OpenAPITools/openapi-generator/pull/14159

**Comment by wtrocki** [comment]
Yes. It will be good for the problem to not reoccur. Example/integration test will guarantee that

**Comment by wtrocki** [comment]
> If a model is an AllOf or does not have any composed schemas at all, the sub-models are in-lined defined in the struct

IMHO that is core of the problem. Inlining operation does also generate separate  the submodelsAlOf object. Right now they are still generated but not used. Time import will be added properly to inlined resource but missing on the alOf one.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractGoCodegen.java (hunk 1 above)
Comment (icubbon) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractGoCodegen.java:685]: Good catch! looks like I screwed up taking the changes from https://github.com/OpenAPITools/openapi-generator/pull/14057/files

File: samples/openapi3/client/petstore/go/go-petstore/model_all_of_primitive_types_all_of.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+API version: 1.0.0
+*/
+
+// Code generated by OpenAPI Generator (https://openapi-generator.tech); DO NOT EDIT.
+
+package petstore
+
+import (
+	"encoding/json"
+	"time"
```
Comment (wtrocki) [inline samples/openapi3/client/petstore/go/go-petstore/model_all_of_primitive_types_all_of.go:15]: Problem solved. Creating issue to seed disscusion if we need this object

**Comment by wtrocki** [comment]
Awesome work! this in fact fixes all the issues we seen. Created separate bug to remove this dangling file https://github.com/OpenAPITools/openapi-generator/issues/14479

FYI: @antihax @grokify @kemokemo @jirikuncar @ph4r5h4d @wing328

File: samples/openapi3/client/petstore/go/go-petstore/api/openapi.yaml [hunk 3]
```diff
@@ -2176,6 +2179,13 @@ components:
           type: string
       type: object
       example: null
+    AllOfPrimitiveTypes_allOf:
```
Comment (wtrocki) [inline samples/openapi3/client/petstore/go/go-petstore/api/openapi.yaml:2182]: Link

**Review by wtrocki — APPROVED** [review]

**Comment by wtrocki** [comment]
🤞 for fast merge. This change is going to resolve number of github issues for golang generator

**Pull request merged**